packages/compiler/core/checker.ts

```diff
@@ -5270,12 +5284,56 @@ function linkMapper<T extends Type>(typeDef: T, mapper?: TypeMapper) {
   }
 }
 
+function extractMainDoc(type: Type): string | undefined {
+  if (type.node?.docs === undefined) {
+    return undefined;
+  }
+  let mainDoc: string = "";
+  for (const doc of type.node.docs) {
+    mainDoc += getDocContent(doc.content);
+  }
+  return mainDoc;
+}
+
+function extractParamDoc(node: OperationStatementNode, paramName: string): string | undefined {
+  if (node.docs === undefined) {
+    return undefined;
+  }
+  for (const doc of node.docs) {
+    for (const tag of doc.tags) {
+      if (tag.kind === SyntaxKind.DocParamTag && tag.paramName.sv === paramName) {
+        return getDocContent(tag.content);
+      }
+    }
+  }
+  return undefined;
+}
+
+function getDocContent(content: readonly DocContent[]) {
+  const docs = [];
+  for (const node of content) {
+    compilerAssert(
+      node.kind === SyntaxKind.DocText,
+      "No other doc content node kinds exist yet. Update this code appropriately when more are added."
+    );
+    docs.push(node.text);
+  }
+  return docs.join("");
```

Comment:
This is copied from elsewhere IIRC. Should we put it in a shared place?

Reply:
yeah copied form the ref doc generator, let me see if I can make this a reusable helper

Reply:
actually I am not sure I should now, just because when we have `{@link foo}` the result shouldn't be a raw string but something else that might have to dealt with and then changing the api would be breaking. So might just keep this as internal for now with a tiny bit of duplication

Reply:
Yeah, that's fine!